src/codeu/chat/client/commandline/Chat.java

```diff
@@ -481,4 +766,144 @@ public void invoke(List<String> args) {
+      Set<Uuid> conversations = updatedConversationsMap.get(user);
+      conversations.remove(convo);
+      updatedConversationsMap.put(user,conversations);
+    }
+
+  }
+
+  // Increments the message count of a specified conversation of interest
+  // of a specific user.
+  private void incrementMessageCount(Uuid convoID){
+
+    for(UserContext user : rootPanelContext.allUsers()){
+      HashMap<Uuid, Integer> convoCount = (convoMessageCountsMap.get(user.user.id) == null) ? new HashMap<>() : convoMessageCountsMap.get(user.user.id);
+      Integer count = (convoCount.get(convoID) == null) ? 0 : convoCount.get(convoID);
+
+      convoCount.put(convoID, count + 1);
+      convoMessageCountsMap.put(user.user.id, convoCount);
+
+    }
+  }
+
+  // Getter method for conversation message counts in conversations of interest.
+  private Integer getMessageCount(Uuid userID, Uuid convoID){
+    return (convoMessageCountsMap.get(userID) == null) ? 0 : convoMessageCountsMap.get(userID).get(convoID);
+  }
+
+  //methods below are helper methods to get a user or conversation from name or Uuid
+
+  // Find the first user with the given name and return a user context
+  // for that user. If no user is found, the function will return null.
+  private UserContext findUser(String name) {
+    for (final UserContext user : rootPanelContext.allUsers()) {
+      if (user.user.name.equals(name)) {
+        return user;
+      }
+    }
+    return null;
+  }
+
+  // Finds the first user with the given Uuid and returns a user context
+  // for that user. If no user is found, the function will return null.
+  private UserContext findUser(Uuid id) {
+    for (final UserContext user : rootPanelContext.allUsers()) {
+      if (user.user.id.equals(id)) {
+        return user;
+      }
+    }
+    return null;
+  }
+
+  // Find the first conversation with the given name and return its context.
+  // If no conversation has the given name, this will return null.
+  private ConversationContext findConversation(String title) {
+    for (final ConversationContext conversation : userPanelContext.conversations()) {
+      if (title.equals(conversation.conversation.title)) {
+        return conversation;
+      }
+    }
+    return null;
+  }
+
+  // Finds the first conversation with the given name and returns its context.
+  // If no conversation has the given name, this will return null.
+  private ConversationContext findConversation(Uuid id) {
+    for (final ConversationContext conversation : userPanelContext.conversations()) {
+      if (id.equals(conversation.conversation.id)) {
+        return conversation;
+      }
+     }
+    return null;
+   }
+
+  public void addUserInterest(Uuid userID, Uuid followedUserID){
+    Set<Uuid> userInterest = (userInterestMap.get(userID) == null) ? new HashSet<>() : userInterestMap.get(userID);
+
+    // Check if the user is trying to follow themselves
+    if(userID.id() == followedUserID.id()) {
```

Comment:
The `.id()` method in the `Uuid` class returns an `int` so `==` is the only option.